File: app/internal/publisher/template/javascript/widgets/status_widget.js
export class StatusWidget {
  constructor(container, engine) {
    this.container = container;
    this.engine = engine;
    this._icon = null;
    this._label = null;
  }

  build() {
    const div = document.createElement('div');
    div.style.cssText = 'display:flex;align-items:center;gap:6px;padding:4px 8px;font-family:sans-serif;font-size:14px;';

    this._icon = document.createElement('span');
    this._icon.textContent = '\u25CF';
    this._icon.style.color = '#888';
    div.appendChild(this._icon);

    this._label = document.createElement('span');
    this._label.textContent = 'Checking...';
    this._label.style.color = '#555';
    div.appendChild(this._label);

    this.container.appendChild(div);
    this.refresh();
    return div;
  }

  async refresh() {
    const s = this.engine.getStatus() || await this.engine.initialize();
    if (s?.valid) {
      const text = s.trial_active ? `Trial: ${s.days_remaining}d` : `Licensed: ${s.days_remaining}d`;
      const color = s.trial_active ? '#f59e0b' : '#16a34a';
      this._label.textContent = text;
      this._label.style.color = color;
      this._icon.style.color = color;
    } else {
      this._label.textContent = s?.message || 'No license';
      this._label.style.color = '#dc2626';
      this._icon.style.color = '#dc2626';
    }
  }
}

File: app/internal/publisher/template/python/widgets/status_widget.py
"""Status Widget for ${product_name}"""
import tkinter as tk
from tkinter import ttk


class StatusWidget:
    def __init__(self, parent, engine):
        self.parent=parent; self.engine=engine; self.frame=ttk.Frame(parent)
    def build(self)->ttk.Frame:
        self.frame.pack(fill=tk.X,padx=5,pady=2)
        branding=self.engine.config.get('branding',{}); colors=branding.get('colors',{}); labels=branding.get('labels',{})
        bg=colors.get('bg_page','#f8f9fa'); tx_m=colors.get('text_muted','#888'); tx_s=colors.get('text_secondary','#555')
        inner=tk.Frame(self.frame,bg=bg,padx=8,pady=4); inner.pack(fill=tk.X)
        self.icon=tk.Label(inner,text="\u25CF",font=("Helvetica",14),bg=bg,fg=tx_m)
        self.icon.pack(side=tk.LEFT,padx=(0,5))
        self.label=tk.Label(inner,text=labels.get('checking_status',"Checking..."),font=("Helvetica",10),bg=bg,fg=tx_s)
        self.label.pack(side=tk.LEFT)
        self.refresh(); return self.frame
    def refresh(self):
        branding=self.engine.config.get('branding',{}); colors=branding.get('colors',{}); labels=branding.get('labels',{})
        s=self.engine.get_status() or self.engine.initialize()
        if s and s.valid:
            t=f"{labels.get('trial_active_text','Trial')}: {s.days_remaining}d" if s.trial_active else f"{labels.get('licensed_text','Licensed')}: {s.days_remaining}d"
            c=colors.get('warning','#f59e0b') if s.trial_active else colors.get('success','#16a34a')
            self.label.config(text=t,fg=c); self.icon.config(fg=c)
        else:
            self.label.config(text=s.message if s else labels.get('no_license_text',"No license"),fg=colors.get('error','#dc2626'))
            self.icon.config(fg=colors.get('error','#dc2626'))
